### ISSUE TITLE
PiecewiseLinear gives wrong CRPS outside of distribution support

### ISSUE THREAD

**Issue by lostella** [awslabs/gluon-ts#563]
It seems like the PiecewiseLinear's CRPS method does not compute correct results when the input falls outside of the distribution's support. The following example is for a uniform distribution between -1 and 1: I would expect CRPS(-2) = CRPS(2), but that's not the case.

```python
gamma = mx.nd.array([-1.0])
slopes = mx.nd.array([
    [2.0, 2.0]]
)
knot_spacings = mx.nd.array([[0.5, 0.5]])

distr = PiecewiseLinear(
    gamma=gamma,
    slopes=slopes,
    knot_spacings=knot_spacings,
)

print(distr.crps(mx.nd.array([-2.0])))
print(distr.crps(mx.nd.array([2.0])))
```

yields

```
[1.6666667]
<NDArray 1 @cpu(0)>

[2.1666667]
<NDArray 1 @cpu(0)>
```

With more complex example one can also get negative values of CRPS.

```python
gamma = mx.nd.array([-82.70746])
slopes = mx.nd.array([
    [1.8083544e+01, 2.9134894e-02, 6.3528619e+00, 4.8505863e-05, 4.9647682e+01, 3.9714784e-16, 1.3027676e+01, 1.8006723e-14]]
)
knot_spacings = mx.nd.array([[0.001, 0.009, 0.09, 0.4, 0.4, 0.09, 0.009, 0.001]])
x = mx.nd.array([-70.8735308])

distr = PiecewiseLinear(
    gamma=gamma,
    slopes=slopes,
    knot_spacings=knot_spacings,
)

print(distr.crps(x))
```

yields

```
[-79.25251]
<NDArray 1 @cpu(0)>
```

**Comment by lostella** [comment]
We could consider implement the computation of CRPS in its more intuitive form, see first equation in https://journals.ametsoc.org/doi/pdf/10.1175/1520-0434%282000%29015%3C0559%3ADOTCRP%3E2.0.CO%3B2

If all slopes are strictly positive then the quantile function is invertible and maybe things are easier to compute in the CDF domain instead of the quantile level domain.

**Comment by lostella** [comment]
Well, one should probably end up with the same formula for the integral no matter what (equivalent) formulation one starts from. It looks like some summands are missing when x falls off of the support.

**Comment by lostella** [comment]
fixed by #564
